crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -723,16 +722,6 @@ impl<'t> TextEdit<'t> {
                 }
             }
 
-            // Allocate additional space if edits were made this frame that changed the size. This is important so that,
-            // if there's a ScrollArea, it can properly scroll to the cursor.
-            let extra_size = galley.size() - rect.size();
-            if extra_size.x > 0.0 || extra_size.y > 0.0 {
-                ui.allocate_rect(
-                    Rect::from_min_size(outer_rect.max, extra_size),
-                    Sense::hover(),
-                );
-            }
-
```

Comment:
Removing code marked as "important" is not a good look. If you want to change this, please create a test (using egui_kittest) for both the issue described in the comment, and your issue